src/components/GeneralInfo/SystemCard/SystemCard.js

```diff
@@ -10,241 +10,223 @@ import EditButton from '../EditButton';
 import { generalMapper } from '../dataMapper';
 import { extraShape } from '../../../constants';
 
-class SystemCardCore extends Component {
-  state = {
-    isDisplayNameModalOpen: false,
-    isAnsibleHostModalOpen: false,
-  };
+const SystemCardCore = (props) => {
+  const {
+    detailLoaded = false,
+    entity = {},
+    properties = {},
+    setDisplayName,
+    setAnsibleHost,
+    writePermissions,
+    handleClick,
+    hasHostName = true,
+    hasDisplayName = true,
+    hasAnsibleHostname = true,
+    hasWorkspace = true,
+    hasSAP = true,
+    hasSystemPurpose = true,
+    hasCPUs = true,
+    hasSockets = true,
+    hasCores = true,
+    hasCPUFlags = true,
+    hasRAM = true,
+    extra = [],
+  } = props;
+
+  const [isDisplayNameModalOpen, setIsDisplayNameModalOpen] = useState(false);
+  const [isAnsibleHostModalOpen, setIsAnsibleHostModalOpen] = useState(false);
 
-  onCancel = () => {
-    this.setState({
-      isDisplayNameModalOpen: false,
-      isAnsibleHostModalOpen: false,
-    });
+  const onCancel = () => {
+    setIsDisplayNameModalOpen(false);
+    setIsAnsibleHostModalOpen(false);
   };
 
-  onSubmit = (fn, origValue) => (value) => {
-    const { entity } = this.props;
+  const onSubmit = (fn, origValue) => (value) => {
+    const { entity } = props;
```

Comment:
This also can be omitted and we can pick the entity from the outer component's scope already